Communitter/src/main/java/com/communitter/api/service/UserService.java

```diff
@@ -62,5 +77,69 @@ public User updateUserAbout(Long id, String about) {
         return userRepo.save(existingUser);
     }
 
+    @Transactional
+    public List<UserFollowerDto> getFollowersByFollowee(Long followeeId) {
+
+        List<UserFollow> followers = userFollowRepository.findAllByFolloweeId(followeeId);
+
+        return followers.stream().map(userMapper::toUserFollowerDto).toList();
+    }
+
+    @Transactional
+    public List<UserFolloweeDto> getFolloweesByFollower(Long followerId) {
+
+        List<UserFollow> followees = userFollowRepository.findAllByFollowerId(followerId);
+
+        return followees.stream().map(userMapper::toUserFolloweeDto).toList();
+    }
+
+    @Transactional
+    public void follow(User follower, Long followeeId) {
 
+        if(Objects.equals(follower.getId(), followeeId)){
+            throw new RuntimeException("You cannot follow yourself");
+        }
+
+        if (userFollowRepository.findByFollowerIdAndFolloweeId(follower.getId(),
+            followeeId).isPresent()) {
+            throw new RuntimeException("You are already followed this user");
+        }
+
+        User followee = userRepo.findById(followeeId).orElseThrow();
+
+        userFollowRepository.save(
+            userMapper.toUserFollowEntity(follower, followee, new Date()));
+    }
+
+    @Transactional
+    public void unfollow(User follower, Long followeeId) {
+
+        if(Objects.equals(follower.getId(), followeeId)){
+            throw new RuntimeException("You cannot unfollow yourself");
+        }
+
+        UserFollow userFollow = userFollowRepository.findByFollowerIdAndFolloweeId(follower.getId(),
+            followeeId).orElseThrow();
```

Comment:
The error message should be added to being explanatory

Reply:
orElseThrow returns no data fount message automatically.